libvips/iofuncs/type.c

```diff
@@ -563,10 +563,8 @@ vips_ref_string_new(const char *str)
 {
 	VipsArea *area;
 
-	if (!g_utf8_validate(str, -1, NULL))
-		str = "<invalid utf-8 string>";
-
-	area = vips_area_new((VipsCallbackFn) vips_area_free_cb, g_strdup(str));
+	area = vips_area_new((VipsCallbackFn) vips_area_free_cb,
+		g_utf8_make_valid(str, -1));
```

Comment:
Do we need to check for `utf8_make_valid()` returning `NULL`?

Reply:
The only case when `utf8_make_valid` returns `NULL` is when the input string is `NULL`. But `g_utf8_validate' would cause segfault with a `NULL` input, so I guess we believe that all the input strings are not `NULL` here.